std/algorithm/searching.d

```diff
@@ -2361,19 +2361,49 @@ If $(D haystack) is a random-access range, all three components of the
 tuple have the same type as $(D haystack). Otherwise, $(D haystack)
 must be a forward range and the type of $(D result[0]) and $(D
 result[1]) is the same as $(XREF range,takeExactly).
+
+Returns:
+A sub-type of Tuple!() with opCast defined for bool.  This opCast returns $(D
+true) when the separating $(D needle) was found (!result[1].empty) and $(D
+false) otherwise.  This enables the convenient idiom shown in the following
+example.
+
+Example:
+---
+if (const split = haystack.findSplit(needle))
```

Comment:
No, documented unittest means writing a `unittest` block with a ddoc comment preceding it. The compiler will insert the body of the unittest into the generated ddocs. For example:
````
/// My awesome function
auto myFunc(R)(R range) { ... }

///
unittest
{
    // The code here will be part of the generated ddocs.
}
````